Consul.Test/CatalogTest.cs

```diff
@@ -317,5 +317,100 @@ public async Task Catalog_ServicesForNodes()
             Assert.Contains(services.Response.Services, n => n.ID == svcID);
             Assert.DoesNotContain(services.Response.Services, n => n.ID == svcID2);
         }
+
+        [Fact]
+        public async Task Catalog_GatewayServices()
+        {
+            using (IConsulClient client = new ConsulClient(c =>
+            {
+                c.Token = TestHelper.MasterToken;
+                c.Address = TestHelper.HttpUri;
+            }))
+            {
+                var terminatingGatewayName = "terminating-gateway";
```

Comment:
I would use a non-standard name, e.g.:
```suggestion
                var terminatingGatewayName = "my-terminating-gateway";
```